test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+            None => UCode::OUT_OF_RANGE,
+        };
+
+        let _reqid_msb = match value.get("reqid").and_then(|resource| resource.get("msb")) {
+            Some(_reqid_msb) => _reqid_msb
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _reqid_lsb = match value.get("reqid").and_then(|resource| resource.get("lsb")) {
+            Some(_reqid_lsb) => _reqid_lsb
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u64>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let ___reqid = UUID {
+            msb: _reqid_msb,
+            lsb: _reqid_lsb,
+            special_fields: SpecialFields::default(),
+        };
+      
+        let __reqid = MessageField(Some(Box::new(___reqid)));
+
+        let _token = match value.get("token") {
+            Some(_token) => _token
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with token field")),
+            None => "Null",
+        };
+
+        let _traceparent = match value.get("traceparent") {
+            Some(_traceparent) => _traceparent
+                .as_str()
+                .unwrap_or_else(|| panic!("Deserialize: something wrong with traceparen field")),
+            None => "Null",
+        };
+        // special field //todo
+        let _special_fields = SpecialFields::default();
+        let mut _uattributes =  UAttributes::new();
+       if _special_fields.ne(&SpecialFields::default()){
+        _uattributes.special_fields = _special_fields;
+       }
+        _uattributes.id = __id;
+        _uattributes.type_ = _type.into();
+
+        
+        if !(_source.0.clone() == UUri::default()) {
+        _uattributes.source = MessageField(Some(Box::new(_source.0)));
+        }
+        if!(_sink.0.clone() == UUri::default()){
+            _uattributes.sink = MessageField(Some(Box::new(_sink.0)));
+            }
+            _uattributes.priority = _priority.into();
+            _uattributes.ttl = _ttl.into();
+            _uattributes.permission_level = Some(_permission_level.into());
+            _uattributes.commstatus = Some(_commstatus.into());
+            _uattributes.reqid = __reqid;
+            _uattributes.token= Some(_token.to_owned());
+            _uattributes.traceparent = Some(_traceparent.to_owned());
+        
+        Ok(WrapperUAttribute( _uattributes))
+    }
+}
+#[derive(Default)]
+pub struct WrapperUPayload(pub UPayload);
+impl<'de> Deserialize<'de> for WrapperUPayload {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value: Value = Deserialize::deserialize(deserializer)?;
+
+        fn from_str_format(s: &str) -> UPayloadFormat {
```

Comment:
I think the [same suggestion](https://github.com/eclipse-uprotocol/up-tck/pull/23/files#r1566222380) applies here, UPayloadFormat has the same facilities for it via the protobuf crate.

Reply:
corrected